src/Evolution/Systems/Cce/AnalyticSolutions/SphericalMetricData.hpp

```diff
@@ -0,0 +1,148 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/Cce/AnalyticSolutions/WorldtubeData.hpp"
+#include "Evolution/Systems/Cce/BoundaryDataTags.hpp"
+#include "Evolution/Systems/Cce/Tags.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/AnalyticSolution.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class DataVector;
+/// \endcond
+
+namespace Cce {
+namespace Solutions {
+
+/*!
+ * \brief Abstract base class for analytic worldtube data most easily derived in
+ * spherical coordinate form.
+ *
+ * \details This class provides the functions required by the `WorldtubeData`
+ * interface that convert from a spherical coordinate spacetime metric to
+ * Cartesian coordinates. Derived classes of `SphericalMetricData` need not
+ * implement the `variables_impl`s for the Cartesian quantities. Instead, the
+ * derived classes must override the protected functions:
+ * - `SphericalMetricData::spherical_metric()`
+ * - `SphericalMetricData::dr_spherical_metric()`
+ * - `SphericalMetricData::dt_spherical_metric()`
+ * Derived classes are still responsible for overriding
+ * `WorldtubeData::get_clone()`, `WorldtubeData::variables_impl()` for tag
+ * `Cce::Tags::News`, and `WorldtubeData::prepare_solution()`.
+ */
+struct SphericalMetricData : public WorldtubeData {
+  /*!
+   * Computes and returns by pointer the Jacobian
+   * \f$\partial x_{\mathrm{spherical}}^j / \partial
+   * x_{\mathrm{Cartesian}}^i\f$
+   */
+  void inverse_jacobian(gsl::not_null<CartesianiSphericalJ*> inverse_jacobian,
+                        size_t l_max) const noexcept;
+
+  /*!
+   * Computes and returns by pointer the first radial derivative of the
+   * Jacobian: \f$\partial_r (\partial x_{\mathrm{spherical}}^j / \partial
+   * x_{\mathrm{Cartesian}}^i)\f$
+   */
+  void dr_inverse_jacobian(
+      gsl::not_null<CartesianiSphericalJ*> dr_inverse_jacobian,
+      size_t l_max) const noexcept;
+
+ protected:
+  using WorldtubeData::variables_impl;
+
+  /*!
+   * \brief Computes the Cartesian spacetime metric from the spherical solution
+   * provided by the derived classes.
+   *
+   * \details The derived classes provide spherical metric data via the virtual
+   * function `SphericalMetricData::spherical_metric()` at a resolution
+   * determined by member variable `l_max_`. This function performs the
+   * coordinate transformation using the Jacobian computed from
+   * `SphericalMetricData::inverse_jacobian()` and performs a re-grid to the
```

Comment:
what is the reason for re-gridding? (I.e. why are there two resolutions?)

Reply:
Sometimes the internal computations, for instance the robinson-trautman solution, or the angular derivatives needed to assemble the spatial derivatives of the spatial metric, are operations that are sensitive to the angular resolution.
So, my reasoning was that calling code might prefer to have a different resolution for the boundary calculation than for the main evolution, though now that you point it out, it probably makes the interface more confusing and gives only marginal gains in efficiency.
tl;dr: I'll remove the resampling. :P